src/components/questionnaire/questionnaire-table.tsx

```diff
@@ -0,0 +1,155 @@
+
+/**
+ * Questionnaire Table Component
+ * @returns Questionnaires from DynamoDB rendered in a x-data-grid table
+ */
+const QuestionnaireTable = () => {
+  const adminMode = UserRoleUtils.adminMode();
+  const { questionnaireApi } = useLambdasApi();
+  const [questionnaires, setQuestionnaires] = useState<Questionnaire[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [deleteId, setDeleteId] = useState<string | null>(null);
+  const [deleteTitle, setDeleteTitle] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchQuestionnaires = async () => {
+      setLoading(true);
+      try {
+        const questionnaires = await questionnaireApi.listQuestionnaires();
+        setQuestionnaires(questionnaires);
+      } finally {
+        setLoading(false);
+      }
+    };
+    fetchQuestionnaires();
+  }, []);
+
+  const handleOpenDialog = (id: string, title: string) => {
+    setDeleteId(id);
+    setDeleteTitle(title);
+    setDialogOpen(true);
+  };
+
+  const handleCloseDialog = () => {
+    setDialogOpen(false);
+    setDeleteId(null);
+    setDeleteTitle(null);
+  };
+
+  const handleConfirmDelete = async () => {
+    if (!deleteId) return;
+    setLoading(true);
+    try {
+      await questionnaireApi.deleteQuestionnaire({ id: deleteId });
+      setQuestionnaires((prevQuestionnaires: Questionnaire[]) =>
+        prevQuestionnaires.filter((questionnaire: Questionnaire) => questionnaire.id !== deleteId)
+      );
+      handleCloseDialog();
+    } catch (error) {
+      console.error("Delete Questionnaire failed:", error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const columns = [
+    { field: "title", headerName: `${strings.questionnaireTable.title}`, flex: 3 },
+    { field: "description", headerName: `${strings.questionnaireTable.description}`, flex: 5 },
+    adminMode
+      ? {
+          field: "actions",
+          headerName: `${strings.questionnaireTable.actions}`,
+          flex: 2.5,
+          renderCell: (params: GridRenderCellParams) => (
+            <>
+              <Button variant="outlined" color="success" sx={{ mr: 1 }}>
+                <EditIcon sx={{ color: "success.main", mr: 1 }} />
+                {strings.questionnaireTable.edit}
+              </Button>
+              <Button
+                variant="contained"
+                color="secondary"
+                onClick={() => handleOpenDialog(params.row.id, params.row.title)}
+              >
+                <DeleteForeverIcon sx={{ color: "red", mr: 1 }} />
+                {strings.questionnaireTable.delete}
+              </Button>
+            </>
+          )
+        }
+      : {
+          field: "status",
+          headerName: `${strings.questionnaireTable.status}`,
+          flex: 1,
+          renderCell: (params: GridRenderCellParams) =>
+            params.row.passedUsers && params.row.passedUsers.lenght > 0 ? (
```

Comment:
Am I correct in thinking we want the check if the logged in user is a passed user? Isn't this rendering the check if there is any passedUser? If so this should be checking if loggedInUser.id is in passedUsers

Reply:
Yep, this was left behind from when mockdata was rendered. I think this should be its own PR in the future. Where this status checking is added to this.